core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -5643,7 +5642,7 @@ object ZIO extends ZIOCompanionPlatformSpecific with ZIOCompanionVersionSpecific
     def apply[A](
       f: Service => A
     )(implicit tagged: Tag[Service], trace: Trace): ZIO[Service, Nothing, A] =
-      ZIO.serviceWithZIO(service => ZIO.succeed(f(service)))
+      ZIO.serviceWithZIO(service => Exit.succeed(f(service)))
```

Comment:
```suggestion
      ZIO.serviceWith(f)
```

Reply:
This method is the implementation of `ZIO.serviceWith`. If we do this we'll end up with infinite recursion

Reply:
of course 😅